services/director-v2/src/simcore_service_director_v2/api/routes/computations.py

```diff
@@ -155,39 +147,64 @@ async def _check_pipeline_startable(
-        return {}
-
-    parent_node_id = NodeID(current_project_metadata["node_id"])
-    parent_node_idstr = NodeIDStr(f"{parent_node_id}")
     try:
-        parent_project_id = await project_repo.get_project_id_from_node(parent_node_id)
-        parent_project = await project_repo.get_project(parent_project_id)
-        assert parent_node_idstr in parent_project.workbench
+        project_ancestors = await projects_metadata_repo.get_project_ancestors(
+            project_id
+        )
+        if project_ancestors.parent_project_uuid is None:
+            # no parents here
+            return {}
+
+        assert project_ancestors.parent_node_id is not None  # nosec
+        assert project_ancestors.root_project_uuid is not None  # nosec
+        assert project_ancestors.root_node_id is not None  # nosec
+
+        async def _get_project_node_names(
+            project_uuid: ProjectID, node_id: NodeID
+        ) -> tuple[str, str]:
+            prj = await project_repo.get_project(project_uuid)
+            node_id_str = NodeIDStr(f"{node_id}")
+            if node_id_str not in prj.workbench:
+                _logger.error(
+                    "%s not found in %s. it is an ancestor of %s. Please check!",
+                    f"{node_id=}",
+                    f"{prj.uuid=}",
+                    f"{project_id=}",
+                )
+                return prj.name, _UNKNOWN_NODE
+            return prj.name, prj.workbench[node_id_str].label
+
+        parent_project_name, parent_node_name = await _get_project_node_names(
+            project_ancestors.parent_project_uuid, project_ancestors.parent_node_id
+        )
+        root_parent_project_name, root_parent_node_name = await _get_project_node_names(
+            project_ancestors.root_project_uuid, project_ancestors.root_node_id
+        )
         return ProjectMetadataDict(
-            parent_node_id=parent_node_id,
-            parent_node_name=parent_project.workbench[parent_node_idstr].label,
-            parent_project_id=parent_project_id,
-            parent_project_name=parent_project.name,
+            parent_node_id=project_ancestors.parent_node_id,
+            parent_node_name=parent_node_name,
+            parent_project_id=project_ancestors.parent_project_uuid,
+            parent_project_name=parent_project_name,
+            root_parent_node_id=project_ancestors.root_node_id,
+            root_parent_node_name=root_parent_node_name,
+            root_parent_project_id=project_ancestors.root_project_uuid,
+            root_parent_project_name=root_parent_project_name,
         )
-    except (
-        ProjectNotFoundError,
-        ProjectNodesNodeNotFoundError,
-        ProjectNodesNonUniqueNodeFoundError,
-    ):
-        _logger.exception("Could not find project/node: %s", f"{parent_node_id=}")
-        return {}
+
+    except DBProjectNotFoundError:
+        _logger.exception("Could not find project: %s", f"{project_id=}")
```

Comment:
TIP: When exceptions are well defined (e.g. inherit from `OsparcErrorMixin`, well defined msg_template, etc ) this handlers would simplify further as 
```python
try
   #...
except (DBProjectNotFoundError,ProjectNotFoundError ) as err:
        _logger.exception("Fatal error %s. See further details for diagnostics", err.code, extras=...)
```

Reply:
alright. will try to remember that OsparcErrorMixin for next times